src/test/java/io/openliberty/tools/intellij/it/TestUtils.java

```diff
@@ -562,4 +571,28 @@ public static void copyDirectory(String sourceDirectoryLocation, String destinat
                 });
     }
 
+    /**
+     * Test to see if there has been a fatal error and JUnit should be stopped.
+     * This searches the output of this JUnit run for SocketTimeoutException which
+     * has been identified as a fatal error and occurs during the Mac tests.
+     */
+    public static synchronized void detectFatalError() {
```

Comment:
After moving the Reader's lifecycle into this method you don't need to synchronize this method. You may want to change this later, but wouldn't worry about doing it in this PR.

Reply:
I will do this in a future update.